### PR TITLE
[VC] syncer: introduce meta cluster

### PR DESCRIPTION
**Pull request by zhuangqh**
it allows vc-syncer deploy on external cluster

Signed-off-by: zhuangqh <zhuangqhc@gmail.com>

### CODE REVIEW

File: incubator/virtualcluster/cmd/syncer/app/options/options.go [hunk 1]
```diff
@@ -150,13 +159,61 @@ func BindFlags(l *syncerconfig.SyncerLeaderElectionConfiguration, fs *pflag.Flag
...
+	var (
+		metaRestConfig, restConfig, inClusterConfig *restclient.Config
+		deployOnMetaCluster                         = o.DeploymentClusterName == constants.DeploymentClusterMeta
+		err                                         error
+	)
+	restConfig, err = getClientConfig(c.ComponentConfig.ClientConnection, o.SuperClusterAddress, !deployOnMetaCluster)
+	if err != nil {
+		return nil, err
+	}
+	if deployOnMetaCluster || o.MetaClusterClientConnection.Kubeconfig != "" {
```
Comment (Fei-Guo) [inline incubator/virtualcluster/cmd/syncer/app/options/options.go:184]: Shouldn't this be && not || ?

File: incubator/virtualcluster/cmd/syncer/app/options/options.go [hunk 2]
```diff
@@ -263,16 +311,15 @@ func getInClusterNamespace() (string, error) {
 	return string(namespace), nil
 }
 
-// createClients creates a meta cluster kube client and a super master custer client from the given config and masterOverride.
-func createClients(config componentbaseconfig.ClientConnectionConfiguration, masterOverride string, timeout time.Duration) (clientset.Interface,
-	vcclient.Interface, clientset.Interface, *restclient.Config, error) {
+// getClientConfig creates a meta cluster kube client and a super master custer client from the given config and masterOverride.
```
Comment (Fei-Guo) [inline incubator/virtualcluster/cmd/syncer/app/options/options.go:314]: Please change the comment accordingly.

File: incubator/virtualcluster/cmd/syncer/app/options/options.go [hunk 3]
```diff
@@ -156,7 +164,47 @@ func (o *ResourceSyncerOptions) Config() (*syncerappconfig.Config, error) {
...
-	leaderElectionClient, virtualClusterClient, superMasterClient, restConfig, err := createClients(c.ComponentConfig.ClientConnection, o.SuperMaster, c.ComponentConfig.LeaderElection.RenewDeadline.Duration)
+	var (
+		metaRestConfig, restConfig, inClusterConfig *restclient.Config
+		err                                         error
+	)
+	restConfig, err = getClientConfig(c.ComponentConfig.ClientConnection, o.SuperClusterAddress, !o.DeployOnMetaCluster)
+	if err != nil {
+		return nil, err
+	}
+	if o.DeployOnMetaCluster || o.MetaClusterClientConnection.Kubeconfig != "" {
```
Comment (Fei-Guo) [inline incubator/virtualcluster/cmd/syncer/app/options/options.go:175]: How about removing o.MetaClusterClientConnection.Kubeconfig != "" ?

File: incubator/virtualcluster/cmd/syncer/app/options/options.go [hunk 4]
```diff
@@ -156,7 +164,47 @@ func (o *ResourceSyncerOptions) Config() (*syncerappconfig.Config, error) {
 	c.ComponentConfig = o.ComponentConfig
 
 	// Prepare kube clients
-	leaderElectionClient, virtualClusterClient, superMasterClient, restConfig, err := createClients(c.ComponentConfig.ClientConnection, o.SuperMaster, c.ComponentConfig.LeaderElection.RenewDeadline.Duration)
+	var (
+		metaRestConfig, restConfig, inClusterConfig *restclient.Config
+		err                                         error
+	)
+	restConfig, err = getClientConfig(c.ComponentConfig.ClientConnection, o.SuperClusterAddress, !o.DeployOnMetaCluster)
```
Comment (Fei-Guo) [inline incubator/virtualcluster/cmd/syncer/app/options/options.go:171]: How about renaming restConfig to superRestConfig? 
How about removing inClusterConfig and use leaderElectionRestConfig directly?

Comment (zhuangqh) [reply incubator/virtualcluster/cmd/syncer/app/options/options.go]: fixed

**Pull request merged**